URL param support for redis adapter

G0V1NDS opened the pull request:
This should be better alternative to #129

tpope commented:
This is indeed a better approach. But we should limit it to connection parameters; half or more of these (e.g., `eval`) have no business being in a connection URI. And we don't need single character aliases.

G0V1NDS commented:
@tpope I just went through the `redis-cli --help` to get the non-boolean flags and added here. 
You are right we can limit to only connection parameters. I think few flags like cert, key, cacert, capath and tls-ciphers should be sufficient. I might be missing some crucial flags here but we can add those later when there is a need of it.

I added single character flag support because few of the flag are single character but not alias like `u,r,i,d,D and c`.

G0V1NDS commented:
@tpope Have added the suggested changes, please review once again and let me know if any other change is required.

tpope commented:
Didn't notice those single character options weren't aliases. But they don't really necessary to me, with the possible exception of `-c`. I would drop all of them; if someone wants `-c` we can add it back.

G0V1NDS commented:
@tpope Removed the other single char flags (u, r, i, d and D) and weren't explicitly handling the -c anyways. Now we are good to go.

tpope commented on autoload/db/adapter/redis.vim:
```diff
@@ -7,7 +7,23 @@ function! db#adapter#redis#canonicalize(url) abort
+    " Specifying only connection releated flag here, missing flags can be added later
+    if k =~# '^\%(cert\|key\|cacert\|capath\|tls-ciphers\|tls-ciphersuites\)$' && v isnot# 1
+      call add(cmd, '--' . k . '=' . v)
+    elseif v =~# '^[1Tt]$'
```
This would make sense to have a `k` check too, no? That is `k ==# 'tls' && v =~# '^[1Tt]'`, or something similar.

tpope replied:
Is there an upside to doing it that way? The downside would be it could potentially pass a non-Boolean option as Boolean, resulting in a potentially confusing error message.

G0V1NDS replied:
No upside, just some laziness in handling it 😉 , will add the parameters mentioned in help doc explicitly.

hiberabyss commented:
Options like `no-auth-warning` should be set globally. Set it for every connection is tiresome.

tpope commented:
> ```
>   --no-auth-warning  Don't show warning message when using password on command
>                      line interface.
> ```

Reading this description, yes, the fix for _this_ option isn't a configuration option _or_ a URL parameter, we should just pass it by default (or better, fix it to pass the password as an environment variable).

hiberabyss commented:
Pass password by environment is a good idea. The mysql warning could also be suppressed via environment password like `MYSQL_PWD=pass mysql -u user -h host`.

G0V1NDS commented:
@tpope bumping it once again, apologies for spamming. Just wanted to make sure if it being checked.

tpope commented on doc/dadbod.txt:
```diff
@@ -169,6 +169,9 @@ Redis ~
 Redis doesn't have a username, so use a dummy one in the URL if a password is
 required.
 
+Query parameters such as `c`, `cert`, `key`, `cacert`, `capath`, `tls-ciphers` and
```
This line is too long. Use `textwidth=78`, as seen in the modeline.

G0V1NDS replied:
Corrected

tpope commented on autoload/db/adapter/redis.vim:
```diff
@@ -7,7 +7,25 @@ function! db#adapter#redis#canonicalize(url) abort
+    elseif k ==# 'c' && v =~# '^[1Tt]$'
+      " Some non-alias single char flags like `-c` needs to be passed
+      " with single hyphen `-` char
+      if len(k) == 1
```
This check is redundant, as `k ==# 'c'` implies `len(k) == 1`.

tpope commented on autoload/db/adapter/redis.vim:
```diff
@@ -7,7 +7,25 @@ function! db#adapter#redis#canonicalize(url) abort
+    " Specifying only connection releated flag here, missing flags can be added later
+    if k =~# '^\%(cert\|key\|cacert\|capath\|tls-ciphers\|tls-ciphersuites\)$' && v isnot# 1
+      call add(cmd, '--' . k . '=' . v)
+    elseif k ==# 'c' && v =~# '^[1Tt]$'
```
Drop the `$` so that `c=true` can be used.

tpope commented on autoload/db/adapter/redis.vim:
```diff
@@ -7,7 +7,25 @@ function! db#adapter#redis#canonicalize(url) abort
+      else
+        call add(cmd, '--' . k)
+      endif
+    else
+      throw 'DB: unsupport URL param `' . k . '` in URL ' . a:url . ', Check `:help dadbod-redis`'
```
Drop this.
